### ISSUE TITLE
Nu LSP crashes on encountering unicode values

### ISSUE THREAD

**Issue by KAAtheWiseGit** [nushell/nushell#12745]
### Describe the bug

The LSP `nu --lsp`, used by Neovim, crashes when it encounters non-ASCII Unicode values.

### How to reproduce

```
1. `Ctrl+o` to open a Neovim editor with Nushell LSP enabled.
2. type `path ä<space>`.
3. LSP crashes.
```

### Expected behavior

Not crash.

### Configuration

| key                | value                                                        |
| ------------------ | ------------------------------------------------------------ |
| version            | 0.92.0                                                       |
| branch             |                                                              |
| commit_hash        |                                                              |
| build_os           | linux-x86_64                                                 |
| build_target       | x86_64-alpine-linux-musl                                     |
| rust_version       | rustc 1.77.0 (aedd173a2 2024-03-17) (Alpine Linux 1.77.0-r0) |
| cargo_version      | cargo 1.77.0                                                 |
| build_time         | 2024-04-03 10:33:03 +00:00                                   |
| build_rust_channel | release                                                      |
| allocator          | mimalloc                                                     |
| features           | default, sqlite, trash, which                                |
| installed_plugins  |                                                              |

### Additional context

I'm unsure how to properly log the LSP output.  If there's a way, I can replicate the bug and submit the logs.

Neovim itself outputs:

```
Error detected while processing InsertCharPre Autocommands for "*":                                                                                                 
Error executing lua callback: /usr/share/nvim/runtime/lua/vim/lsp.lua:1980: attempt to index local 'client' (a nil value)
stack traceback:
        /usr/share/nvim/runtime/lua/vim/lsp.lua:1980: in function 'cancel_fun'
        ...m/site/pack/deps/start/mini.nvim/lua/mini/completion.lua:805: in function 'cancel_lsp'
        ...m/site/pack/deps/start/mini.nvim/lua/mini/completion.lua:740: in function <...m/site/pack/deps/start/mini.nvim/lua/mini/completion.lua:738>
```

Which, I presume, means that the LSP client has disconnected.

**Comment by fdncred** [comment]
@schrieveslaach Would you mind looking into this?

@KAAtheWiseGit I'm not sure about logging either but my first step would be to look at the testing. Seems like we had quite a few tests for the lsp.

**Comment by schrieveslaach** [comment]
@KAAtheWiseGit, I couldn't reproduce the issue with the Arch builds or with the latest development version.

| key                | value                                                            |
| ------------------ | ---------------------------------------------------------------- |
| version            | 0.92.2                                                           |
| branch             | makepkg                                                          |
| commit_hash        | 2a08a18b26865a86c793d183b9b042220ecb733a                         |
| build_os           | linux-x86_64                                                     |
| build_target       | x86_64-unknown-linux-gnu                                         |
| rust_version       | rustc 1.77.1 (7cf61ebde 2024-03-27) (Arch Linux rust 1:1.77.1-2) |
| cargo_version      | cargo 1.77.1                                                     |
| build_time         | 2024-04-11 08:40:50 +00:00                                       |
| build_rust_channel | release                                                          |
| allocator          | mimalloc                                                         |
| features           | dataframe, default, sqlite, trash, which                         |
| installed_plugins  |                                                                  |

I'm wondering if this is a `x86_64-unknown-linux-gnu` vs `x86_64-alpine-linux-musl` thing. :thinking: Is there some CI tests with a musl based system?

@fdncred is right, [there are tests with utf8](https://github.com/nushell/nushell/blob/0d6fbdde4adc927c62549f8f75d15a87044a10b7/crates/nu-lsp/src/lib.rs#L1143).

**Comment by KAAtheWiseGit** [comment]
I thought it might've been a Neovim-specific thing, but I can't replicate it with `gopls` or Rust analyzer.

**Comment by sholderbach** [comment]
Worth keeping in mind that the [default encoding for positions specified by the protocol](https://microsoft.github.io/language-server-protocol/specifications/lsp/3.17/specification/#positionEncodingKind) is UTF-16 until the capability negotiation is done.

**Comment by galuszkak** [comment]
@kaathewisegit  @sholderbach - wasn't this fixed in #14742 ?

**Comment by fdncred** [comment]
Let me know if it wasn't and i'll reopen.

**Comment by kaathewisegit** [comment]
Yep, I can confirm it works with Nu 0.110.0 in both arguments and identifiers
